### DISCUSSION TITLE
Invent Sapphire Framework?

### DISCUSSION THREAD

**Discussion by tintinkung** [ASEAN-Build-The-Earth/Merlion#25] [Ideas]
I'm inspecterizin Merlion and i found theres sapphire framework [installed.(V7.17.0)](https://github.com/ASEAN-Build-The-Earth/Merlion/tree/main/node_modules/%40sapphire/async-queue)
> ![sapphire_module](https://user-images.githubusercontent.com/77855014/136762208-239c669c-f5ab-4192-959c-875aa3ac77b5.png)
> ![sapphire_ver](https://user-images.githubusercontent.com/77855014/136762213-30f4ceb6-2486-4c66-8ef1-5291ea7e1212.png)

---
### so how about consider using it?
*pros:*
 - better command handler(i think)
 - better long-term organizing
 - TypeScript supports
 - full docomentation at [sapphire site](https://sapphiredev.github.io/utilities/index.html)
 - open source framework(well of course) at [github](https://github.com/sapphiredev/framework)

*cons:*
 - for better performance, gotta learn typescript
 - gotta learn framework
 - go read docomentation
 - I've no idea how to set it up

*example:*
Method | Basic ping pong command
------------ | -------------
Default | ![ping_default](https://user-images.githubusercontent.com/77855014/136763662-27e3e0fe-e533-49a5-8471-b823aa576f4d.png)
Sapphire framework | ![ping_sapphire](https://user-images.githubusercontent.com/77855014/136763659-a7902036-374d-49ea-8f22-477a5b9d96b8.png)

*see example project: [here](https://github.com/sapphiredev/examples/tree/main/examples/with-javascript)*

**Comment by StoneMcYT** [comment]
Shouldn't we use discussions for this?

**Comment by StoneMcYT** [comment]
This is better now

**Comment by StoneMcYT** [comment]
@tintinkung  from what im seeing on Sapphire doc it seems like basically, Sapphire is very complicated? I suppose we can save Saphire to be added and whatnot after we finish updating the bot to v13 discord.js

**Reply by tintinkung** [reply]
yes it is very complicate and simply gotta rework the bot for it. but its very cool with long term organizing.
tbh its just some kul choice for kul project upgrade

**Reply by StoneMcYT** [reply]
> yes it is very complicate and simply gotta rework the bot for it. but its very cool with long term organizing. tbh its just some kul choice for kul project upgrade

Makes sense but at the moment it seems like a lot to do so maybe once we fully upgrade to v13.

**Reply by tintinkung** [reply]
v13 is just syntax error though i think nothing will be harz, gonna finish v13 upgrade pretty fast(if devs active ;-;)

**Reply by StoneMcYT** [reply]
Devs be devin alot sorta? issue rn is that there was accident when dev commit v13 to main branch so like the v12 code is v13/v12 code
